editor/src/clj/editor/resource.clj

```diff
@@ -81,6 +81,10 @@
   (and (openable-resource? value)
        (true? (:editor-openable (resource-type value)))))
 
+(defn openable-in-view-type?
+  [value view-type]
+  (contains? (into #{} (map :id (:view-types (resource-type value)))) view-type))
```

Comment:
I think it would be better to use `(some #(= view-type %) (:view-types ...))`: it is still O(n), but without creating an intermediate set that will be immediately thrown away. We also have `coll/some` to reduce GC pressure even more.